pkg/model/labels_test.go

```diff
@@ -68,6 +68,42 @@ func TestLabelsUnique(t *testing.T) {
 	}
 }
 
+func Test_LabelsBuilder_Unique(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    Labels
+		add      Labels
+		expected Labels
+	}{
+		{
+			name: "duplicates in input are preserved",
+			input: Labels{
+				{Name: "unique", Value: "yes"},
+				{Name: "unique", Value: "no"},
+			},
+			add: Labels{
+				{Name: "foo", Value: "bar"},
+				{Name: "foo", Value: "baz"},
+			},
+			expected: Labels{
+				{Name: "foo", Value: "baz"},
+				{Name: "unique", Value: "yes"},
+				{Name: "unique", Value: "no"},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			builder := NewLabelsBuilder(test.input)
+			for _, l := range test.add {
+				builder.Set(l.Name, l.Value)
+			}
+			assert.Equal(t, test.expected, builder.Labels())
+		})
+	}
+}
+
```

Comment:
I do not see any practical sense in fixing `LabelsBuilder` we borrowed from prometheus. Instead, in our case, a reusable map would work way better (and I'm 99.(9)% sure, more efficiently).

To be addressed in a separate PR.

<img width="553" alt="image" src="https://github.com/grafana/pyroscope/assets/12090599/fdf88226-e4d3-497a-9484-b0f33eab0b3b">